formats/protobuf/commonTest/src/kotlinx/serialization/protobuf/ProtobufTypeParameterTest.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.protobuf
+
+import kotlinx.serialization.*
+import kotlin.test.*
+
+class ProtobufTypeParameterTest {
+    @Serializable
+    data class Box<T>(val value: T)
+
+    @Serializable
+    data class ExplicitNullableUpperBoundBox<T : Any?>(val value: T)
+
+    @Serializable
+    data class ExplicitNullableUpperNullablePropertyBoundBox<T : Any?>(val value: T?)
```

Comment:
Do you mind also adding test for `class Foo<T: Any>(val t: T)` to check that non-nullable type parameters still reject `null`s?

Reply:
I'm sorry I am not sure how to do this. I could add some tests for `class Foo<T: Any>(val t: T)` but it's impossible to set a `null` for the member `t`, unless I go for some hacks like calling the constructor in Java and passing a `null` argument (I am not completely sure this works though).

Reply:
Yeah, you're right, that's impossible for Kotlin's type system. Never mind :)